src/coordinator/storage/local/storage.go

```diff
@@ -65,14 +68,64 @@ func (s *localStorage) Fetch(ctx context.Context, query *storage.FetchQuery, opt
 		return nil, err
 	}
 
-	opts := storage.FetchOptionsToM3Options(options, query)
+	// NB(r): Since we don't use a single index we fan out to each
+	// cluster that can completely fulfill this range and then prefer the
+	// highest resolution (most fine grained) results.
+	// This needs to be optimized, however this is a start.
+	var (
+		opts       = storage.FetchOptionsToM3Options(options, query)
+		namespaces = s.clusters.ClusterNamespaces()
+		now        = time.Now()
+		fetches    = 0
+		result     multiFetchResult
+		wg         sync.WaitGroup
+	)
+	for _, namespace := range namespaces {
+		namespace := namespace // Capture var
+
+		clusterStart := now.Add(-1 * namespace.Attributes().Retention)
```

Comment:
instead of this, can you just check if !clusterStart.Before(query.Start) {...}

Reply:
I want to include the case where query.Start matches the clusterStart however, which that logic would not cover.

Reply:
So `!clusterStart.Before(query.Start) { continue }` is equivalent to `clusterStart.Equal(query.Start) || clusterStart.After(query.Start) { continue }` which is undesirable here.